### ISSUE TITLE
"Quick lyrics had trouble reading your music" banner is always shown

### ISSUE THREAD

**Issue by movalex** [QuickLyric/QuickLyric#237]
Huge red banner "Quick lyrics had trouble reading your music" is showed on my Xiaomi Mi5 when I run Quicklyrics.  All necessary permissions are set. Pressing "Fix it" and re-enabling autostart and storage permissions does nothing. 
By the way, why in a  world Quicklyrics has to have access to my storage if I mostly play streaming music?
Please remove this nasty banner.

**Comment by geecko86** [comment]
Regarding the storage permission:

QuickLyric needs this permission precisely to know if your music comes from a file or from a streaming service. 

And if it does come from a file (in your case it doesn't), it can also extract id3 tags or an audio fingerprint so it can better identify the song. But first it needs to know if it's from a file or not, which is why it needs this permission.

The storage permission is *not* related to the red banner. It's absolutely optional, though it helps a lot if it's enabled and you're using MP3 files. Which, you're not.

---------------------------------------------------------------

Regarding the red banner:

"Autostart" (it has different names depending on the manufacturer) is at the heart of this issue. This problem is specific to Chinese manufacturers like Oppo, Xiaomi, Honor, etc. who enjoy killing background services. Whitelisting QuickLyric is supposed to solve this, but then again, it's very hard for us to tackle this issue, given that we don't have any of those devices. So we're essentially working in the dark.

There is also no API or system tool we can use to know if the app has been granted autostart or not.

The first problem is that we don't have any of these devices here, and that whenever we try to work with a user who does and who is reporting this issue, they usually don't care. At all. They typically send us an angry email and that's it. Meanwhile we're left with nothing.

@movalex This means that the only way we can solve this is if you give us a hand. Please tell us:

1. If Autostart is always enabled or if your Xiaomi phone disables it by itself
2. If you have anything on your device (i.e. Greenify) that might prevent QuickLyric from running in the background.
3. If the red banner has always been there
4. If a reboot fixes the issue?

**Comment by movalex** [comment]
Hi! Thank you for quick answer.
And thanks to your suggestion I was able to fix it. 
1. Autostart was not enabled, and I had to enable it.
2. No such apps
3. The red banner was there from the first time I launched the app.
4. Seems like reboot fixed the issue indeed.
First I checked Xiaomi power management. By default it has battery saver turned on for every app, and it unloads the app from the memory after a period of time. I disabled powermanagement for the app, then rebooted the phone and the banner disappeared. 
I tried to reproduce the bug, but the banner did not appear when I changed power options back. So it seems the autostart options was not applied until reboot. 
By the way, when this banner is shown, it hides top 4 lines of the lyrics. And _this_ seems like a bug.
P.S. Thank you for this gorgeous app. I am particularly tired of Musixmatch's 
poor quality of community driven lyrics :) Definitely going to subscribe QuickLyrics.
